consumer.go

```diff
@@ -255,6 +256,11 @@ type PartitionConsumer interface {
 	// errors are logged and not returned over this channel. If you want to implement any custom errpr
 	// handling, set your config's Consumer.Return.Errors setting to true, and read from this channel.
 	Errors() <-chan *ConsumerError
+
+	// HighWaterMarkOffset returns the high water mark offset of the partition, i.e. the offset that will
+	// be used for the next message that will be produced. You can use this to determine how far behind
+	// the processing is.
+	HighWaterMarkOffset() int64
```

Comment:
Personally, I have a problem with it being **high-water mark offset**; I'd have preferred **high-water mark** or **high-water offset** instead of this Frankensteinian composition. But that would be deviating from the protocol spec too much. Oh well :P